### PR TITLE
Dom/traveler dashboard

### PR DESCRIPTION
**Pull request by dreardon1021**
#### What's this PR do?
Populates the cards needed based on the user logging in
#### Where should the reviewer start?
style sheets, index.js, domUpdates.js
#### How should this be manually tested?
log in with multiple users to see what occurs
#### Questions:
I want the cards to have a background image of the image shown. But couldnt figure out how. Any ideas?

### CODE REVIEW

File: src/domUpdates.js [hunk 1]
```diff
@@ -13,8 +14,81 @@ const domUpdates = {
...
+    $('.display-money').text(`You Spent $${traveler.calculateUserSpent(travelerID, tripData, destinationData)} This Year`)
+  },
+
+  populateUpcomingTrips: (traveler, travelerID, tripData, destinationData) => {
+    let userTrips = traveler.findUserUpcomingTrips(travelerID, tripData)
+    userTrips.forEach(trip => {
+      let foundDestination = destinationData.find(destination => destination.id === trip.destinationID)
+      $('.upcoming-trip-cards').append(`
+        <div class="trip-card">
+          <img src=${foundDestination.image} class="card-background" alt="${foundDestination.destination}" />
```
Comment (kimalajoy) [inline src/domUpdates.js]: This is the one that you are trying to get to display the image as background that you were asking about? Does the alt generate as it should, at least? Or do neither work?

File: src/domUpdates.js [hunk 2]
```diff
@@ -13,8 +14,81 @@ const domUpdates = {
...
 
+  //If cannot be tested yet. Need to do a POST
+  populateTravelerInfo: (traveler, travelerID, tripData, destinationData) => {
+    if (traveler.findCurrentTrips(travelerID, tripData)) {
+      let tripDestination = destinationData.find(destination => traveler.findCurrentTrips(travelerID, tripData).destinationID === destination.id)
+      $('.display-username').text(`Welcome ${traveler.name}, enjoy ${tripDestination.destination}!`)
+    } else {
+      $('.display-username').text(`Welcome ${traveler.name}!`)
+    }
+    $('.display-username').text(`Welcome ${traveler.name}!`)
```
Comment (kimalajoy) [inline src/domUpdates.js]: I think line 27 will always override the text from the conditional (line 23) so it seems like you have an extra/repeated line

File: src/domUpdates.js (hunk 1 above)
Comment (kimalajoy) [reply src/domUpdates.js]: Maybe it's because you are using img and not background image?

File: src/index.js [hunk 3]
```diff
@@ -69,17 +74,33 @@ function loginUser() {
...
   destinationData.forEach(destination => {
     domUpdates.populateDestinationDropDown(destination)
   })
 }
 
+//helper function in login
+function extractUserId(userInput) {
+  if (userInput.length === 10) {
+    let id = userInput.split('').splice(-2, 2).join('');
+    return userData.find(user => user.id === parseInt(id))
```
Comment (kimalajoy) [inline src/index.js]: You should only need to return one time at the end of this function (you have duplicate code on 101 and 104, so you should be able to just have the one return statement after the curly on line 105

**Comment by kimalajoy** [comment]
Added a few comments about some extra lines that I saw, but I will merge the request so you can keep working! Looking great so far! You are putting in so much hard work on this!

File: src/domUpdates.js (hunk 1 above)
Comment (dreardon1021) [reply src/domUpdates.js]: Yea so I tried using the background image with css but cant figure out how to target the card being created at that time resulting in many images or over writing the background image of all previously generated cards.

File: src/domUpdates.js (hunk 2 above)
Comment (dreardon1021) [reply src/domUpdates.js]: OH! I think youre right

File: src/index.js (hunk 3 above)
Comment (dreardon1021) [reply src/index.js]: Hmmmm I think I see what youre saying here. Establish the variable id and return the statement at the end based upon the conditional

**Pull request merged**